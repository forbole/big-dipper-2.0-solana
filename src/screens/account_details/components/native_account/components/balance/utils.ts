import { formatNumber } from '@utils/format_token';
import { BalanceType } from './types';

export const formatBalanceData = (data: BalanceType) => {
  const balanceChart = [
    {
      key: 'balanceNative',
      display: `${formatNumber(data.native.value, data.native.exponent)} ${data.native.displayDenom.toUpperCase()}`,
      value: data.native.value,
    },
    {
      key: 'balanceStake',
      display: `${formatNumber(data.stake.value, data.stake.exponent)} ${data.stake.displayDenom.toUpperCase()}`,
      value: data.stake.value,
    },
    {
      key: 'balanceNonce',
      display: `${formatNumber(data.nonce.value, data.nonce.exponent)} ${data.nonce.displayDenom.toUpperCase()}`,
      value: data.nonce.value,
    },
    {
      key: 'balanceVote',
      display: `${formatNumber(data.vote.value, data.vote.exponent)} ${data.vote.displayDenom.toUpperCase()}`,
      value: data.vote.value,
    },
  ];

  return balanceChart;
};
