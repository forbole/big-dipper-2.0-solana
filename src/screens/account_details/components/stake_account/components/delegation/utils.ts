import numeral from 'numeral';
import { DelegationType } from './types';

export const formatBalanceData = (data: DelegationType) => {
  const balanceChart = [
    {
      key: 'balanceActivated',
      display: `${numeral(data.activated.value).format(data.activated.format)} ${data.activated.denom.toUpperCase()}`,
      value: data.activated.value,
    },
    {
      key: 'balanceActivating',
      display: `${numeral(data.activating.value).format(data.activating.format)} ${data.activating.denom.toUpperCase()}`,
      value: data.activating.value,
    },
  ];

  return balanceChart;
};
