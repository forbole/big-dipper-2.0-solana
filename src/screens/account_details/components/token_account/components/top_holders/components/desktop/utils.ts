export const columns:{
  key: string;
  align?: 'left' | 'center' | 'right' | 'justify' | 'inherit';
  width: number;
}[] = [
  {
    key: 'rank',
    width: 5,
    align: 'center',
  },
  {
    key: 'holder',
    width: 65,
  },
  {
    key: 'amount',
    width: 30,
    align: 'right',
  },
];
