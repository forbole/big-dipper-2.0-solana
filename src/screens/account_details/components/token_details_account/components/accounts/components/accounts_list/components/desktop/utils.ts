export const columns:{
  key: string;
  align?: 'left' | 'center' | 'right' | 'justify' | 'inherit';
  width: number;
}[] = [
  {
    key: 'address',
    width: 95,
  },
];
