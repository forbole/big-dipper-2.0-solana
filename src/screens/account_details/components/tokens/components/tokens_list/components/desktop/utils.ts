export const columns:{
  key: string;
  align?: 'left' | 'center' | 'right' | 'justify' | 'inherit';
  width: number;
}[] = [
  {
    key: 'token',
    width: 50,
  },
  {
    key: 'amount',
    align: 'right',
    width: 50,
  },
];
