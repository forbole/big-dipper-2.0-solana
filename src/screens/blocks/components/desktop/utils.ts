export const columns:{
  key: string;
  align?: 'left' | 'center' | 'right' | 'justify' | 'inherit';
  width: number;
}[] = [
  {
    key: 'slot',
    width: 10,
  },
  {
    key: 'leader',
    width: 35,
  },
  {
    key: 'hash',
    width: 30,
  },
  {
    key: 'txs',
    align: 'right',
    width: 10,
  },
  {
    key: 'time',
    align: 'right',
    width: 15,
  },
];
