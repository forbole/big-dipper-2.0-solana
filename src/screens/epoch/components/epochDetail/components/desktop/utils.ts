export const columns:{
  key: string;
  align?: 'left' | 'center' | 'right' | 'justify' | 'inherit';
  width: number;
}[] = [
  {
    key: 'name',
    width: 30,
  },
  {
    key: 'value',
    width: 20,
  },
  {
    key: 'detail',
    width: 50,
  },
];
