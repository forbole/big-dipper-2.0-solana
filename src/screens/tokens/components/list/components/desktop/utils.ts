export const columns:{
  key: string;
  align?: 'left' | 'center' | 'right' | 'justify' | 'inherit';
  width: number;
  component?: React.ReactNode;
  sortKey?: string;
  sort?: boolean;
}[] = [
  {
    key: 'idx',
    width: 10,
  },
  {
    key: 'token',
    width: 30,
    sort: true,
    sortKey: 'token',
  },
  {
    key: 'price',
    width: 20,
    align: 'right',
    sort: true,
    sortKey: 'price',
  },
  {
    key: 'marketCap',
    align: 'right',
    width: 20,
    sort: true,
    sortKey: 'marketCap',
  },
  {
    key: 'volume',
    align: 'right',
    width: 20,
    sort: true,
    sortKey: 'marketCap',
  },
];
