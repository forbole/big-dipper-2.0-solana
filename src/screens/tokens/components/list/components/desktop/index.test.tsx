import React from 'react';
import renderer from 'react-test-renderer';
import { MockTheme } from '@tests/utils';
import Desktop from '.';

// ==================================
// mocks
// ==================================
const mockI18n = {
  t: (key: string) => key,
  lang: 'en',
};
jest.mock('next-translate/useTranslation', () => () => mockI18n);
jest.mock('@components', () => ({
  Loading: (props) => <div id="Loading" {...props} />,
  SortArrows: (props) => <div id="SortArrows" {...props} />,
}));

jest.mock('react-virtualized-auto-sizer', () => ({ children }: any) => children({
  height: 600, width: 600,
}));

// ==================================
// unit tests
// ==================================
describe('screen: Home/Blocks/Desktop', () => {
  it('matches snapshot', () => {
    const component = renderer.create(
      <MockTheme>
        <Desktop
          items={[
            {
              token: 'BTC',
              price: 1802,
              marketCap: 515912496,
              address: '9n4nbM75f5Ui33ZbPYXn59EwSgE8CGsHtAeTH5YFeJ9E',
            },
          ]}
          itemCount={1}
          loadMoreItems={() => jest.fn()}
          isItemLoaded={() => true}
          handleSort={jest.fn()}
          sortDirection="asc"
          sortKey="token"
        />
      </MockTheme>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  afterEach(() => {
    jest.clearAllMocks();
  });
});
