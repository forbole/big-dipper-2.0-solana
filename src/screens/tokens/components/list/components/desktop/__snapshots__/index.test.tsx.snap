// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screen: Home/Blocks/Desktop matches snapshot 1`] = `
<div
  className="makeStyles-root"
>
  <div
    onScroll={[Function]}
    style={
      Object {
        "WebkitOverflowScrolling": "touch",
        "direction": "ltr",
        "height": 50,
        "overflow": "auto",
        "position": "relative",
        "width": 600,
        "willChange": "transform",
      }
    }
  >
    <div
      style={
        Object {
          "height": 50,
          "pointerEvents": undefined,
          "width": 600,
        }
      }
    >
      <div
        className="makeStyles-cell"
        onClick={[Function]}
        role="button"
        style={
          Object {
            "height": 50,
            "left": 0,
            "position": "absolute",
            "right": undefined,
            "top": 0,
            "width": 60,
          }
        }
      >
        <h4
          className="MuiTypography-root MuiTypography-h4"
        >
          idx
        </h4>
      </div>
      <div
        className="makeStyles-cell makeStyles-flexCells undefined sort"
        onClick={[Function]}
        role="button"
        style={
          Object {
            "height": 50,
            "left": 60,
            "position": "absolute",
            "right": undefined,
            "top": 0,
            "width": 180,
          }
        }
      >
        <h4
          className="MuiTypography-root MuiTypography-h4"
        >
          token
          <div
            id="SortArrows"
            sort="asc"
          />
        </h4>
      </div>
      <div
        className="makeStyles-cell makeStyles-flexCells right sort"
        onClick={[Function]}
        role="button"
        style={
          Object {
            "height": 50,
            "left": 240,
            "position": "absolute",
            "right": undefined,
            "top": 0,
            "width": 120,
          }
        }
      >
        <h4
          className="MuiTypography-root MuiTypography-h4 MuiTypography-alignRight"
        >
          price
          <div
            id="SortArrows"
          />
        </h4>
      </div>
      <div
        className="makeStyles-cell makeStyles-flexCells right sort"
        onClick={[Function]}
        role="button"
        style={
          Object {
            "height": 50,
            "left": 360,
            "position": "absolute",
            "right": undefined,
            "top": 0,
            "width": 120,
          }
        }
      >
        <h4
          className="MuiTypography-root MuiTypography-h4 MuiTypography-alignRight"
        >
          marketCap
          <div
            id="SortArrows"
          />
        </h4>
      </div>
      <div
        className="makeStyles-cell makeStyles-flexCells right sort"
        onClick={[Function]}
        role="button"
        style={
          Object {
            "height": 50,
            "left": 480,
            "position": "absolute",
            "right": undefined,
            "top": 0,
            "width": 120,
          }
        }
      >
        <h4
          className="MuiTypography-root MuiTypography-h4 MuiTypography-alignRight"
        >
          volume
          <div
            id="SortArrows"
          />
        </h4>
      </div>
    </div>
  </div>
  <div
    className="scrollbar"
    onScroll={[Function]}
    style={
      Object {
        "WebkitOverflowScrolling": "touch",
        "direction": "ltr",
        "height": 550,
        "overflow": "auto",
        "position": "relative",
        "width": 600,
        "willChange": "transform",
      }
    }
  >
    <div
      style={
        Object {
          "height": 50,
          "pointerEvents": undefined,
          "width": 600,
        }
      }
    >
      <div
        className="makeStyles-cell makeStyles-body odd"
        style={
          Object {
            "height": 50,
            "left": 0,
            "position": "absolute",
            "right": undefined,
            "top": 0,
            "width": 60,
          }
        }
      >
        <div
          className="MuiTypography-root MuiTypography-body1"
        >
          #1
        </div>
      </div>
      <div
        className="makeStyles-cell makeStyles-body odd"
        style={
          Object {
            "height": 50,
            "left": 60,
            "position": "absolute",
            "right": undefined,
            "top": 0,
            "width": 180,
          }
        }
      >
        <div
          className="MuiTypography-root MuiTypography-body1"
        >
          <div
            address="9n4nbM75f5Ui33ZbPYXn59EwSgE8CGsHtAeTH5YFeJ9E"
            href={[Function]}
            id="AvatarName"
            imageUrl={null}
            name="BTC"
          />
        </div>
      </div>
      <div
        className="makeStyles-cell makeStyles-body odd"
        style={
          Object {
            "height": 50,
            "left": 240,
            "position": "absolute",
            "right": undefined,
            "top": 0,
            "width": 120,
          }
        }
      >
        <div
          className="MuiTypography-root MuiTypography-body1 MuiTypography-alignRight"
        >
          $1,802
        </div>
      </div>
      <div
        className="makeStyles-cell makeStyles-body odd"
        style={
          Object {
            "height": 50,
            "left": 360,
            "position": "absolute",
            "right": undefined,
            "top": 0,
            "width": 120,
          }
        }
      >
        <div
          className="MuiTypography-root MuiTypography-body1 MuiTypography-alignRight"
        >
          $515,912,496
        </div>
      </div>
      <div
        className="makeStyles-cell makeStyles-body odd"
        style={
          Object {
            "height": 50,
            "left": 480,
            "position": "absolute",
            "right": undefined,
            "top": 0,
            "width": 120,
          }
        }
      >
        <div
          className="MuiTypography-root MuiTypography-body1 MuiTypography-alignRight"
        >
          $18,333,060
        </div>
      </div>
    </div>
  </div>
</div>
`;
